Makefile

```diff
@@ -122,6 +122,10 @@ else
 	@# NOTE: supports TEST_FILE=test_foo.res to run test_foo actually
 	+$(SINGLE_MAKE) -C src/nvim/testdir NVIM_PRG="$(realpath build/bin/nvim)" $(patsubst %.res,%,$(TEST_FILE)) SCRIPTS= $(MAKEOVERRIDES)
 endif
+# Build oldtest by specifying the .vim filename.
+.PHONY: phony_force
+src/nvim/testdir/%: phony_force
+	$(MAKE) oldtest TEST_FILE=$(patsubst src/nvim/testdir/%.vim,%,$@)
```

Comment:
this is nice, but it would be better to have `TEST_FILE=foo.vim` working instead, so that we can have simpler documentation and consistent usage.

Reply:
The rule here handles "src/…/test_foo.vim" => TEST_FILE=test_foo (removing the .vim).
The ".res" above is substituted for B/C.
`make src/nvim/testdir/test_environ.res` does not work.  It could be made to work, but it is better to not have it.
I will change the pattern above to only handle `%.vim`.

Reply:
> `make src/nvim/testdir/test_environ.res` does not work. It could be made to work, but it is better to not have it.

I'm not asking for `make src/nvim/testdir/test_environ.res` to work. I meant `TEST_FILE=foo.vim` specifically. Currently (on master) this works:

    TEST_FILE=foo.res make oldtest

But this would be better (more intuitive):

        TEST_FILE=foo.vim make oldtest

Reply:
Ok, should work now also.  It is a bit unfortunate that "TEST_FILE" is used here.

After all `make -C src/nvim/testdir test_foo` now also works - but I've kept it to go through the old rule for dependencies etc.  But it also makes it very noisy.

Reply:
> but I've kept it to go through the old rule for dependencies etc. But it also makes it very noisy.

...which appears to be not working properly anymore for the TSan build.